plugins/autoid/plugin.js

```diff
@@ -1,9 +1,95 @@
 'use strict';
 
 (function () {
-	CKEDITOR.plugins.add('autoid', {
-		init: function (editor) {
+  var editorHasFocus = false;
+  var commandIsActive = false;
```

Comment:
I know this pattern is cloned from nanospell, but I wonder if it's kind of an antipattern since commands already have state.  http://docs.ckeditor.com/#!/api/CKEDITOR.command-property-state